contracts/src/FLK.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import "solmate/tokens/ERC20.sol";
+
+contract FleekToken is ERC20 {
+    /* STATE */
+
+    mapping(address => bool) private _minters;
+
+    /* EVENTS */
+
+    event MinterAdded(address indexed account);
+    event MinterRemoved(address indexed account);
+
+    constructor(uint256 _initialSupply) ERC20("Fleek", "FLK", 18) {
+        _mint(msg.sender, _initialSupply);
+
+        _addMinter(msg.sender);
+    }
+
+    /**
+     * @dev Add a new minter.
+     * @param _account Address of the minter
+     */
+    function addMinter(address _account) external {
+        _addMinter(_account);
+    }
+
+    /**
+     * @dev Remove a minter.
+     * @param _account Address of the minter
+     */
+    function removeMinter(address _account) external {
+        _removeMinter(_account);
+    }
+
+    /**
+     * @dev Mint new tokens.
+     * @param _to Address to send the newly minted tokens
+     * @param _amount Amount of tokens to mint
+     */
+    function mint(address _to, uint256 _amount) external {
+        _mint(_to, _amount);
```

Comment:
```suggestion
        require(isMinter(_to), "Only minters can mint new tokens");
        _mint(_to, _amount);
```
Should we check if the receiving address is a minter or is that not necessary?

Reply:
Removed access control from this since this contract isnt integrated into protocol at all yet, and pending transaction verification spec. Was put in for beards tests. Might be worth keeping out of this PR in this state though.